ext/curl/multi.c

```diff
@@ -187,7 +187,13 @@ PHP_FUNCTION(curl_multi_select)
 
 	mh = Z_CURL_MULTI_P(z_mh);
 
-	error = curl_multi_wait(mh->multi, NULL, 0, (unsigned long) (timeout * 1000.0), &numfds);
+	if (!(timeout >= 0.0 && timeout <= ((double)INT_MAX / 1000.0))) {
+		php_error_docref(NULL, E_WARNING, "timeout must be between 0 and %d", (int)ceilf((double)INT_MAX / 1000));
```

Comment:
Sorry, the `ceilf()` is still not quite correct. Maybe we should actually output a float value? Or maybe leave as is, and call it a day. It's quite unlikely that a user specifies a timeout of almost a month.